### PR TITLE
Add tree-sitter-nix

### PR DESCRIPTION
**Pull request by nrdxp**
Basic syntax highlighting and indentation should work. I couldn't get rnix-lsp to behave properly, so it is not included for now. Will add if I can figure it out before this gets merged.

### CODE REVIEW

File: runtime/queries/nix/indents.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+indent = [
+ "if",
+ "let",
+ "function",
+ "attrset",
+ "list",
+ "indented_string",
+ "parenthesized"
+]
```
Comment (archseer) [inline runtime/queries/nix/indents.toml:9]: You likely want to define some of the brackets as outdents, we don't automatically outdent when typing those yet, but it's used in certain calculations (like if you calculated the indent of every existing line)

**Comment by archseer** [comment]
`rnix-lsp` what was the issue there? We might be missing some capabilities in the LSP client

File: runtime/queries/nix/indents.toml (hunk 1 above)
Comment (nrdxp) [reply runtime/queries/nix/indents.toml:9]: I actually removed the outdents because it seemed to screw up the indents (by not indenting enough spaces)

**Comment by nrdxp** [comment]
In terms of the lsp, I'm not really sure what the issue is so far. According to the logs, it loads just fine, but it doesn't actually do anything. I was able to get diagnostics to show up by creating a syntax error, saving the file, and reloading it, but the diagnostics don't show on file change as they normally should. I also tried nix-eval-lsp but that didn't work either. Both projects are nowhere near feature complete though, so it may not be Helix' fault.

**Comment by archseer** [comment]
I'll merge this as is, we can improve the LSP support & extend the indent queries later :)

**Comment by 573** [comment]
@nrdxp I had a similar problem with vim+rnix-lsp. After I set filetype=nix, it just worked.

**Pull request merged**